doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+  registration code in :file:`myapp/__init__.py`. 
+
+* :file:`myapp/plugin.py` contains the code that defines the physics plugin we will create
+
+* :file:`myapp/fluxSolver.py` contains the flux solver
+
+* :file:`myapp/thermalSolver.py` contains the thermal/hydraulics solver
+
+* :file:`setup.py` the python package installation file to help users install your
+  application.
+
+* :file:`README.md` and :file:`LICENSE.md` are an optional description and license of your
+  application that would be prominently featured, e.g. in a github repo, if you were to
+  put it there.
+
+Registering the app with ARMI
+=============================
+The ARMI Framework contains features to run the "main loop" of typical applications. In
+order to get access to these, we must register our new app with the ARMI framework. To do
+this, we put the following code in the top-level :file:`__init__.py` module:
+
+.. code-block:: python
+    :caption: ``myapp/__init__.py``
+
+    import armi
+    from myapp import app
+    armi.configure(app.ExampleApp())
+
+Defining the app class
+======================
+We define our app in the :file:`myapp/app.py` module. For this example, the app class is
+relatively simple: it will just register our one custom plugin. We will actually create
+the plugin shortly.
+
+.. admonition:: Apps vs. plugins vs. interfaces
+
+    ARMI-based methodologies are broken down into three layers of abstraction. Apps are
+    collections of plugins intended to perform analysis on a certain type of reactor.
+    Plugins are independent and mixable collections of relatively arbitrary code that
+    might bring in special materials, contain certain engineering methodologies, and/or
+    Interfaces with one or more physics kernels.
+
+.. code-block:: python
+    :caption: ``myapp/app.py``
+
+    import armi
+    from armi.apps import App
+
+    from myapp.plugin import DummyPhysicsPlugin
+
+    class ExampleApp(App):
+        def __init__(self):
+            # activate all built-in plugins
+            App.__init__(self)
+
+            # register our plugin with the plugin manager
+            self._pm.register(DummyPhysicsPlugin) 
+
+        @property
+        def splashText(self):
+            return "** My Example App **"
+
+
+Defining the physics plugin
+===========================
+Now we will create the plugin that will coordinate our dummy physics modules. 
+
+.. admonition:: What are plugins again? 
+
+    Plugins are the basic modular building block of ARMI-based apps. In some cases, one plugin
+    will be associated with one physics kernel (like MCNP). This is a reasonable practice when
+    you expect to be mixing and matching various combinations of plugins between related
+    teams. It is also possible to have a plugin that performs a whole cacophony of analyses
+    using multiple codes, which some smaller research teams may find preferable. The
+    flexibility is very broad.
```

Comment:
There may be some room here to mention some guidance about separating plugins based on being able to distribute/install the software too. Sometimes it makes sense to lump everything into a single plugin, but if you want to share small chunks of code it may be better to make more plugins from the get-go.

Reply:
Beyond the scope of the tutorial, and goes waaaay off into documenting Python territory, where I don't really think we belong. I think we should maybe provide a link to the python docs that discuss modules/packages (Nick did this) and leave it at that. I think we could be more clear and concise and provide all the documentation that *we* should be providing by stating "ARMI plugins are just subclasses of the ArmiPlugin class. You can put them wherever you like, so long as you can import them where needed."